base/save.go

```diff
@@ -200,9 +200,18 @@ func CreateDataset(ctx context.Context, r repo.Repo, streams ioes.IOStreams, ds,
 	ds.ProfileID = pro.ID.String()
 	ds.Peername = pro.Peername
 	ds.Path = path
-	if err = r.Logbook().WriteVersionSave(ctx, ds); err != nil && err != logbook.ErrNoLogbook {
+	action, err := r.Logbook().WriteVersionSave(ctx, ds)
+	if err != nil && err != logbook.ErrNoLogbook {
 		return
 	}
+	dscache := r.Dscache()
+	if dscache != nil && !dscache.IsEmpty() {
+		log.Info("dscache: update and save new version")
+		err = dscache.Update(action)
+		if err != nil {
+			log.Error(err)
+		}
+	}
```

Comment:
These two things must always be called together. We should keep an eye out for other places where two subsystems have required interactions like this. Lots of this PR is feeling like another step toward an event bus

Reply:
I'm going to keep this as in for now, but I see the problem. I'll change this over to using a channel in the next dscache PR.